tests/suites/helpers.function

```diff
@@ -163,18 +130,16 @@ typedef struct data_tag
  *
  * \param   TEST                The test expression to be tested.
  */
-#define TEST_INVALID_PARAM_RET( PARAM_ERR_VALUE, TEST ) \
-    do {                                                \
-        if ( setjmp( param_fail_jmp ) == 0 )            \
-        {                                               \
-            if( (TEST) != PARAM_ERR_VALUE)              \
-            {                                           \
-                test_fail( #TEST, __LINE__, __FILE__ ); \
-                goto exit;                              \
-            }                                           \
-       }                                                \
-       memset( param_fail_jmp, 0, sizeof(jmp_buf) );    \
-    } while( 0 )
+#define TEST_INVALID_PARAM_RET( PARAM_ERR_VALUE, TEST )                     \
+    do {                                                                    \
+        test_info.paramfail_test_state = PARAMFAIL_TESTSTATE_PENDING;       \
+        if( (TEST) != (PARAM_ERR_VALUE) &&                                  \
+            test_info.paramfail_test_state != PARAMFAIL_TESTSTATE_CALLED )  \
```

Comment:
This condition seems a bit vague. I would expect that we expect both: result to be as expected _and_ the failure callback to be called, which would mean we use `||` here.

Reply:
Good catch! I noticed independently while doing some manual testing, but didn't immediately understand what was going wrong, until @hanno-arm pointed me to your comment which had the answer I was looking for. Thanks!